src/build.cpp

```diff
@@ -65,29 +70,53 @@ void buildInternal(Database* db, const Entry& e,
 
     LOGD << "Relative path " << relativePath;
 
-    try{
+    try {
         io::hardlink(relativePath, hardlink);
         LOGD << "Linked " << relativePath << " --> " << hardlink;
         relativePath = hardlink;
-    }catch(const FSException &e){
+    } catch(const FSException &e){
         LOGD << e.what();
         // Will build directly from path
     }
 
-
     // We could vectorize this logic, but it's an overkill by now
-    try{
+    try {
         if (e.type == PointCloud) {
             const std::vector vec = {relativePath};
 
-            buildEpt(vec, o);
+            buildEpt(vec, tempFolder);
+
+            LOGD << "Build complete, moving temp folder to " << outputFolder;
+            io::assureFolderExists(fs::path(outputFolder).parent_path());
+            std::filesystem::rename(tempFolder, outputFolder);
 
-            output << o << std::endl;
+            LOGD << "Temp folder moved";
+
+            output << outputFolder << std::endl;
         }
 
+        io::assureIsRemoved(tempFolder);
         io::assureIsRemoved(hardlink);
-    }catch(...){
+    
+    // Catch block is now detailed because we want to keep track of the exception
+    // catches pdal, gdal, filesystem_error, etc...
+    } catch(const std::exception& ex){
+
+        LOGD << ex.what();
+
+        io::assureIsRemoved(tempFolder);
         io::assureIsRemoved(hardlink);
+
+        throw;
+
+    // This way we are sure that all the exceptions are caught
+    } catch(const std::string& ex){
```

Comment:
Which code throws a string? Throwing strings in general is a really bad idea. https://isocpp.org/wiki/faq/exceptions#what-to-throw

Reply:
Fixed in https://github.com/DroneDB/DroneDB/pull/265/commits/99bc8ebf63a8559f98c614023be6da18a043a34c